src/modules/hip/kernel/gaussian_filter.hpp

```diff
@@ -0,0 +1,2065 @@
+        rpp_hip_adjust_range(dstPtr, &sum_f24);
+        rpp_hip_pack_float24_pln3_and_store24_pkd3(dstPtr + dstIdx, &sum_f24);
+    }
+}
+
+// kernelSize = 9
+template <typename T>
+__global__ void gaussian_filter_9x9_pkd_tensor(T *srcPtr,
+                                               uint2 srcStridesNH,
+                                               T *dstPtr,
+                                               uint2 dstStridesNH,
+                                               uint padLength,
+                                               uint2 tileSize,
+                                               RpptROIPtr roiTensorPtrSrc,
+                                               d_float81 *filterTensor)
+{
+    int hipThreadIdx_x8 = hipThreadIdx_x << 3;
+    int id_x_o = (hipBlockIdx_x * tileSize.x * 8) + hipThreadIdx_x8;
+    int id_y_o = hipBlockIdx_y * tileSize.y + hipThreadIdx_y;
+    int id_z = hipBlockIdx_z * hipBlockDim_z + hipThreadIdx_z;
+
+    int id_x_i = id_x_o - padLength;
+    int id_y_i = id_y_o - padLength;
+    d_float24 sum_f24;
+    __shared__ uchar src_lds[48][128];
+
+    int srcIdx = (id_z * srcStridesNH.x) + ((id_y_i + roiTensorPtrSrc[id_z].xywhROI.xy.y) * srcStridesNH.y) + ((id_x_i + roiTensorPtrSrc[id_z].xywhROI.xy.x) * 3);
+    int dstIdx = (id_z * dstStridesNH.x) + (id_y_o * dstStridesNH.y) + id_x_o * 3;
+    d_float81 filter_f81 = filterTensor[id_z];
+    sum_f24.f4[0] = (float4) 0;
+    sum_f24.f4[1] = (float4) 0;
+    sum_f24.f4[2] = (float4) 0;
+    sum_f24.f4[3] = (float4) 0;
+    sum_f24.f4[4] = (float4) 0;
+    sum_f24.f4[5] = (float4) 0;
+
+    int3 hipThreadIdx_y_channel;
+    hipThreadIdx_y_channel.x = hipThreadIdx_y;
+    hipThreadIdx_y_channel.y = hipThreadIdx_y + 16;
+    hipThreadIdx_y_channel.z = hipThreadIdx_y + 32;
+
+    uchar *src_lds_channel[3];
+    src_lds_channel[0] = &src_lds[hipThreadIdx_y_channel.x][hipThreadIdx_x8];
+    src_lds_channel[1] = &src_lds[hipThreadIdx_y_channel.y][hipThreadIdx_x8];
+    src_lds_channel[2] = &src_lds[hipThreadIdx_y_channel.z][hipThreadIdx_x8];
+
+    if ((id_x_i >= -(int)padLength) && (id_x_i < roiTensorPtrSrc[id_z].xywhROI.roiWidth) &&
+        (id_y_i >= 0) && (id_y_i < roiTensorPtrSrc[id_z].xywhROI.roiHeight))
+    {
+        rpp_hip_load24_pkd3_to_uchar8_pln3(srcPtr + srcIdx, src_lds_channel);
+    }
+    else
+    {
+        *(uint2 *)src_lds_channel[0] = (uint2)0;
+        *(uint2 *)src_lds_channel[1] = (uint2)0;
+        *(uint2 *)src_lds_channel[2] = (uint2)0;
+    }
+    __syncthreads();
+    if ((id_x_o < roiTensorPtrSrc[id_z].xywhROI.roiWidth) &&
+        (id_y_o < roiTensorPtrSrc[id_z].xywhROI.roiHeight) &&
+        (hipThreadIdx_x < tileSize.x) &&
+        (hipThreadIdx_y < tileSize.y))
+    {
+        gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x    ][hipThreadIdx_x8], &sum_f24.f8[0], &filter_f81, 0);
```

Comment:
Please initialize a common temporary variable holding the pointer to current row needed in the filter to not have the 50 odd adds inside gaussian_filter_9x9_row_hip_compute. Same for other sizes.
Something like:
```
d_float9 *filterRow_f9 = (d_float9 *)&filter_f81;
gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x    ][hipThreadIdx_x8], &sum_f24.f8[0], filterRow_f9);
gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y    ][hipThreadIdx_x8], &sum_f24.f8[1], filterRow_f9);
gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z    ][hipThreadIdx_x8], &sum_f24.f8[2], filterRow_f9++);
gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.x + 1][hipThreadIdx_x8], &sum_f24.f8[0], filterRow_f9);
gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.y + 1][hipThreadIdx_x8], &sum_f24.f8[1], filterRow_f9);
gaussian_filter_9x9_row_hip_compute(&src_lds[hipThreadIdx_y_channel.z + 1][hipThreadIdx_x8], &sum_f24.f8[2], filterRow_f9++);
and so on..
```

Reply:
Done this for 3x3 kernels. Need to replicate for other kernel sizes
https://github.com/sampath1117/rpp/blob/sr/opt_gauss_filter_hip/src/modules/hip/kernel/gaussian_filter.hpp#L6
https://github.com/sampath1117/rpp/blob/sr/opt_gauss_filter_hip/src/modules/hip/kernel/gaussian_filter.hpp#L302
https://github.com/sampath1117/rpp/blob/sr/opt_gauss_filter_hip/src/modules/hip/kernel/gaussian_filter.hpp#L339

Reply:
Replicated for the remaining kernel sizes